internal/pkg/service/stream/definition/repository/branch/branch_create.go

```diff
@@ -0,0 +1,55 @@
+package branch
+
+import (
+	"context"
+	"time"
+
+	"github.com/keboola/go-utils/pkg/deepcopy"
+
+	serviceError "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition"
+)
+
+// Create a branch (that already exists in the Storage API) in the Stream Service database.
+// - If there is a deleted Branch with the same key, the Undelete operation is performed.
+// - If the Branch already exists, the ResourceAlreadyExistsError is returned.
+// - If the MaxBranchesPerProject limit is exceeded, the CountLimitReachedError is returned.
+func (r *Repository) Create(input *definition.Branch, now time.Time, by definition.By) *op.AtomicOp[definition.Branch] {
```

Comment:
**Each commit is one repository operation, for example `Create` with test and database snapshot, so review should be really easy.**